### PR TITLE
[BEAM-4802] Update the contribution guide (Dependency section)

### PR DESCRIPTION
**Pull request by yifanzou**
Update the Dependency Guide. Added the Auto-Jira section.
+R: @charlesccychen 
cc: @chamikaramj 

Once your pull request has been opened and assigned a number, please edit the
URL below, replacing `PULL_REQUEST_NUMBER` with the number of your pull request.

http://apache-beam-website-pull-requests.storage.googleapis.com/499/index.html

Finally, it will help us expedite review of your Pull Request if you tag
someone (e.g. @username) to look at it.

### CODE REVIEW

File: src/contribute/dependencies.md [hunk 1]
```diff
@@ -44,14 +44,25 @@ One common solution for the diamond dependency problem is [semantic versioning](
...
 
 In addition to this, Beam community members might identify other critical dependency updates that have to be manually performed. For example,
 * A minor release of a dependency due to a critical security vulnerability. 
 * A dependency conflict that was was triggered by a minor version release of a Beam dependency (this does not apply to Java SDK that depends on exact minor versions of dependencies).
 
 These kind of urgently required upgrades might not get automatically picked up by the Jenkins job for few months. So Beam community has to act to identify such issues and perform upgrades early.
 
+## JIRA Automation
+
+In order to track the dependency upgrade process, JIRA tickets will be created per significant outdated dependency base on the report. A bot named *Beam Jira Bot* was created for managing JIRA issues. Beam community agrees on the following policies that creates and updates issues.
```
Comment (charlesccychen) [inline src/contribute/dependencies.md]: "based"

File: src/contribute/dependencies.md [hunk 2]
```diff
@@ -44,14 +44,25 @@ One common solution for the diamond dependency problem is [semantic versioning](
...
+* Issues will be named as "Beam Dependency Update Request: <dep_name> <dep_newest_version>".
+* Issues will be created under the component *"dependencies"*
+* Issues will be assigned to the primary owner of the dependencies, who are mentioned in the dependency ownership files. ([Java Dependency Owners](https://github.com/apache/beam/blob/master/ownership/JAVA_DEPENDENCY_OWNERS.yaml) and [Python Dependency Owners](https://github.com/apache/beam/blob/master/ownership/PYTHON_DEPENDENCY_OWNERS.yaml))
+* If more than one owners found for a dependency, the first owner will be picked as the primary owner, the others will be pinged in the issue's description.
+* If no owners found, leave the assignee empty. The component lead is responsible for triage the issue.
```
Comment (charlesccychen) [inline src/contribute/dependencies.md]: "triaging"

File: src/contribute/dependencies.md [hunk 3]
```diff
@@ -44,14 +44,25 @@ One common solution for the diamond dependency problem is [semantic versioning](
...
+* Avoid creating duplicate issues. Updating the descriptions of the open issues created by previous dependency updates inspection.
```
Comment (charlesccychen) [inline src/contribute/dependencies.md]: Can you clarify the end of the sentence?  It doesn't quite parse for me, in particular the trailing "updates inspection".

**Comment by charlesccychen** [comment]
@asfgit merge

**Comment by charlesccychen** [comment]
retest this please

**Comment by charlesccychen** [comment]
Looks like there is something wrong with presubmit tooling (CC: @jasonkuster)
```
[beam_PreCommit_Website_Stage] $ /bin/bash -xe /tmp/jenkins5733482887105358262.sh
+ gpg --keyserver hkp://keys.gnupg.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 7D2BAF1CF37B13E2069D6956105BD0E739499BDB
gpg: requesting key D39DC0E3 from hkp server keys.gnupg.net
gpg: requesting key 39499BDB from hkp server keys.gnupg.net
gpg: keyserver timed out
gpg: keyserver receive failed: keyserver error
Build step 'Execute shell' marked build as failure
Setting status of 4ce9d1fa83bb84ba1533cba4c1431e10c1a8a8ec to FAILURE with url https://builds.apache.org/job/beam_PreCommit_Website_Stage/1224/ and message: 'FAILURE
```

**Comment by jasonkuster** [comment]
We see this error periodically -- try rerunning tests?

**Comment by jasonkuster** [comment]
retest this please

**Comment by charlesccychen** [comment]
@asfgit merge

**Pull request closed without merging**